cmds/exp/madeye/doc.go

```diff
@@ -0,0 +1,18 @@
+// u-root was intended to be capable of function as a universal root, i.e. a root
+// file system that you could boot from different architectures.
+// We all this Multiple Architecture Device Image, or MADI, pronounced Mad-Eye.
+// Apologies to Harry Potter.
+// This command implements MADI and works as follows:
+// Given a set of images, e.g.
+// initramfs.linux_<arch>.cpio it derives the architecture from the name.
+// It then reads the cpio in.
+// For all symlinks which resolve to absolute paths, it records the file they point to.
+// For a distinguished set of directories, it relocates them from / to /<arch>/, a la Plan 9.
```

Comment:
so basically, initramfs's are merged.  ones with $ARCH in the name will have their contents extracted to /arch/, with all abs symlinks adjusted?

(not a comment on your doc, more of a question for myself).

this is pretty cool.

Reply:
oh, it looks like just the contents of /bin (and similar) are relocated, and not stuff like lib.  i guess you don't have a lot of stuff in e.g. /usr/lib/libc.so.6 or whatever?

Reply:
/usr and friends are "not yet". This is at the start purely for u-root -> no libc yet.

That's going to have to be done at some point.